### ISSUE TITLE
[Bug]  Tried to use mongodb connector but throwing an error

### ISSUE THREAD

**Issue by rubensacoustic** [ververica/flink-cdc-connectors#2302]
### Search before asking

- [X] I searched in the [issues](https://github.com/ververica/flink-cdc-connectors/issues) and found nothing similar.

### Flink version

1.15

### Flink CDC version

2.4.0

### Database and its version

Mongo Atlas  6.0.6

### Minimal reproduce step

This is the code I'm trying to run, is taken from example.
```
      MongoDBSource<String> mongoSource =
              MongoDBSource.<String>builder()
                      .connectionOptions("retryWrites=true&w=majority&ssl=true")
                      .scheme("mongodb+srv")
                      .hosts("xxxxxx")
                      .databaseList("dbtest") // set captured database, support regex
                      .collectionList("dbtest.config_test") //set captured collections, support regex
                      .username("xxx")
                      .password("xxx")
                      .deserializer(new JsonDebeziumDeserializationSchema())
                      .build();

      StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
      // enable checkpoint
      env.enableCheckpointing(3000);
      // set the source parallelism to 2
      env.fromSource(mongoSource, WatermarkStrategy.noWatermarks(), "MongoDBIncrementalSource")
              .setParallelism(2)
              .print()
              .setParallelism(1);

      env.execute("Print MongoDB Snapshot + Change Stream");

```

### What did you expect to see?

The records flowing throught the pipeline

### What did you see instead?

An exception like this:

com.mongodb.MongoQueryException: Query failed with error code 8000 and error message 'noTimeout cursors are disallowed in this atlas tier' on server dev-ue1-tealeaf-md-shard-00-02.9zymn.mongodb.net:27017

### Anything else?

I see that `noTimeout` cursors are used here:

https://github.com/ververica/flink-cdc-connectors/blob/62b32713a472f7db24bd40ebed9a7e9c04bb4529/flink-connector-mongodb-cdc/src/main/java/com/ververica/cdc/connectors/mongodb/source/reader/fetch/MongoDBScanFetchTask.java#L116

I don't understand well the code base I think this is for  snapshots, Is there a way to avoid using `noTimeout` cursors? As it seems like mongo atlas doesn't support it on my tier.

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

**Comment by yuxiqian** [comment]
Opened #2332 to address this issue. With this patch it should be possible to suppress this behavior like follows:

```diff
      MongoDBSource<String> mongoSource =
              MongoDBSource.<String>builder()
                      .connectionOptions("retryWrites=true&w=majority&ssl=true")
                      .scheme("mongodb+srv")
                      .hosts("xxxxxx")
                      .databaseList("dbtest") // set captured database, support regex
                      .collectionList("dbtest.config_test") //set captured collections, support regex
                      .username("xxx")
                      .password("xxx")
                      .deserializer(new JsonDebeziumDeserializationSchema())
+                     .disableCursorTimeout(false)
                      .build();
```

**Comment by Jiabao-Sun** [comment]
Closed by #2332
